grails-app/services/org/pih/warehouse/core/LocationService.groovy

```diff
@@ -718,4 +723,83 @@ class LocationService {
         }
         existingLocation.delete(flush: true)
     }
+
+    Location createOrUpdateLocation(Map params) {
```

Comment:
@drodzewicz commented that this are just doing data binding so bindLocation is probably more appropriate. And we should move them into the *DataImportService because 1) that's where they are primarily used 2) we can repurpose DataImportService to be generically useful for any type of import / export activities.

Reply:
One concern is that we also do some binding like this on APIs so we might want to make sure that the DataImportService is named appropriately to support other use cases like CRUD operations on the API.

Thoughts:
DataMarshaller, DataMapper - take a look at jackson to see how it handles the API for patterns related to moving data back and forth between string and object.
The other part of this that is complicated related to the conversion from string to object of an primary key, code/identifier, name.